src/components/TodoFilter.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Filter } from '../types/Filter';
+
+type Props = {
+  filterBy: Filter;
+  setFilterBy: (filter: Filter) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filterBy,
+  setFilterBy,
+}) => {
+  return (
+    <nav className="filter" data-cy="Filter">
+      <a
+        href="#/"
+        className={classNames('filter__link', {
+          selected: filterBy === 'all',
+        })}
+        data-cy="FilterLinkAll"
+        onClick={(e) => {
+          e.preventDefault();
+          setFilterBy(Filter.all);
+        }}
+      >
+        All
+      </a>
+
+      <a
+        href="#/active"
+        className={classNames('filter__link', {
+          selected: filterBy === 'active',
+        })}
+        data-cy="FilterLinkActive"
+        onClick={(e) => {
+          e.preventDefault();
+          setFilterBy(Filter.active);
+        }}
+      >
+        Active
+      </a>
+
+      <a
+        href="#/completed"
+        className={classNames('filter__link', {
+          selected: filterBy === 'completed',
+        })}
+        data-cy="FilterLinkCompleted"
+        onClick={(e) => {
+          e.preventDefault();
+          setFilterBy(Filter.completed);
+        }}
+      >
```

Comment:
Do not repet your self also do not use anonymous function in JSX when they more then 1 line
```suggestion
}) => {
  const onFilterChange = (filter: Filter) => (e: Event) => {
    e.preventDefault();
    setFilterBy(filter);
  }

  return (
    <nav className="filter" data-cy="Filter">
      <a
        href="#/"
        className={classNames('filter__link', {
          selected: filterBy === Filter.all,
        })}
        data-cy="FilterLinkAll"
        onClick={onFilterChange(Filter.all)}
      >
        All
      </a>

      <a
        href="#/active"
        className={classNames('filter__link', {
          selected: filterBy === Filter.active,
        })}
        data-cy="FilterLinkActive"
        onClick={onFilterChange(Filter.active)}
      >
        Active
      </a>

      <a
        href="#/completed"
        className={classNames('filter__link', {
          selected: filterBy === Filter.completed,
        })}
        data-cy="FilterLinkCompleted"
        onClick={onFilterChange(Filter.completed)}
      >
```